### ISSUE TITLE
Router connection closed: Slow Consumer And no longer reconnect

### ISSUE THREAD

**Issue by wubumihuo** [cloudfoundry/nats-release#48]
We use nats -v2.1.7，and have three nodes in cluster： 168.63.70.231/233/235
the 231‘s config is :
```
# Monitoring and Tracing
port: 4222
http_port: 8222
# limit
max_pending: 64Mb
write_deadline: 5s
# cluster options
cluster {
  listen: 168.63.70.231:4248
  routes = [
  ]
}
```
and 235's config is ：
```
# Monitoring and Tracing
port: 4222
http_port: 8222

# limit
max_pending: 64Mb
write_deadline: 5s

# cluster options
cluster {
  listen: 168.63.70.235:4248
  routes = [
    nats://168.63.70.231:4248
  ]
}
```
On 2022/10/18 begin at 15:03 ,we found on 168.63.70.231,there are frequent slow consumer on 168.63.70.235 ,the log in below:
```
[99971] 2022/10/18 14:59:53.531838 [DBG] 168.63.70.235:4248 - rid:303 - Router Ping Timer
[99971] 2022/10/18 14:59:53.531868 [DBG] 168.63.70.235:4248 - rid:303 - Delaying PING due to client activity 0s ago
[99971] 2022/10/18 15:01:53.532037 [DBG] 168.63.70.235:4248 - rid:303 - Router Ping Timer
[99971] 2022/10/18 15:01:53.532065 [DBG] 168.63.70.235:4248 - rid:303 - Delaying PING due to client activity 0s ago
[99971] 2022/10/18 15:02:35.853998 [INF] 168.63.70.235:4248 - rid:303 - Slow Consumer Detected: WriteDeadline of 5s exceeded with 197 chunks of 12863290 total bytes.
[99971] 2022/10/18 15:02:40.854214 [INF] 168.63.70.235:4248 - rid:303 - Slow Consumer Detected: WriteDeadline of 5s exceeded with 344 chunks of 22340957 total bytes.
[99971] 2022/10/18 15:02:45.854474 [INF] 168.63.70.235:4248 - rid:303 - Slow Consumer Detected: WriteDeadline of 5s exceeded with 448 chunks of 29152757 total bytes.
[99971] 2022/10/18 15:02:50.854702 [INF] 168.63.70.235:4248 - rid:303 - Slow Consumer Detected: WriteDeadline of 5s exceeded with 518 chunks of 33696862 total bytes.
[99971] 2022/10/18 15:02:55.854911 [INF] 168.63.70.235:4248 - rid:303 - Slow Consumer Detected: WriteDeadline of 5s exceeded with 510 chunks of 33165517 total bytes.
[99971] 2022/10/18 15:03:00.855105 [INF] 168.63.70.235:4248 - rid:303 - Slow Consumer Detected: WriteDeadline of 5s exceeded with 487 chunks of 31562931 total bytes.
[99971] 2022/10/18 15:03:05.855354 [INF] 168.63.70.235:4248 - rid:303 - Slow Consumer Detected: WriteDeadline of 5s exceeded with 496 chunks of 32137173 total bytes.
[99971] 2022/10/18 15:03:05.855381 [INF] 168.63.70.235:4248 - rid:303 - Router connection closed: Slow Consumer (Write Deadline)
[99971] 2022/10/18 15:03:05.861368 [DBG] Attempting reconnect for solicited route "nats-route://168.63.70.235:4248/"
[99971] 2022/10/18 15:03:05.871608 [DBG] Trying to connect to route on 168.63.70.235:4248
[99971] 2022/10/18 15:03:05.879146 [DBG] 168.63.70.235:4248 - rid:16989 - Route connect msg sent
[99971] 2022/10/18 15:03:05.879256 [INF] 168.63.70.235:4248 - rid:16989 - Route connection created
[99971] 2022/10/18 15:03:05.880750 [DBG] 168.63.70.235:4248 - rid:303 - Removing 2088 subscriptions for account "$G"
[99971] 2022/10/18 15:03:06.637781 [DBG] 168.63.70.235:4248 - rid:16989 - Registering remote route "NBZKDSS4KMU7VJKPYWWAJTN226ZZQQ6YFOJT3NA24QACSEAGXIHUBTNT"
[99971] 2022/10/18 15:03:07.174214 [DBG] 168.63.70.235:4248 - rid:16989 - Sent local subscriptions to route
[99971] 2022/10/18 15:03:07.174374 [DBG] 168.63.70.235:4248 - rid:16989 - Router Ping Timer
[99971] 2022/10/18 15:03:12.174436 [INF] 168.63.70.235:4248 - rid:16989 - Slow Consumer Detected: WriteDeadline of 5s exceeded with 2 chunks of 50523370 total bytes.
[99971] 2022/10/18 15:03:17.174653 [INF] 168.63.70.235:4248 - rid:16989 - Slow Consumer Detected: WriteDeadline of 5s exceeded with 4 chunks of 49516271 total bytes.
[99971] 2022/10/18 15:03:22.174853 [INF] 168.63.70.235:4248 - rid:16989 - Slow Consumer Detected: WriteDeadline of 5s exceeded with 8 chunks of 49335771 total bytes.
[99971] 2022/10/18 15:03:22.174885 [INF] 168.63.70.235:4248 - rid:16989 - Router connection closed: Slow Consumer (Write Deadline)
```

and on 168.63.70.235,the log in below:
```
[103465] 2022/10/18 14:59:54.612049 [DBG] 168.63.70.231:47496 - rid:108 - Router Ping Timer
[103465] 2022/10/18 14:59:54.612057 [DBG] 168.63.70.231:47496 - rid:108 - Delaying PING due to client activity 0s ago
[103465] 2022/10/18 15:01:54.861604 [DBG] 168.63.70.231:47496 - rid:108 - Router Ping Timer
[103465] 2022/10/18 15:01:54.866770 [DBG] 168.63.70.231:47496 - rid:108 - Delaying PING due to client activity 0s ago

[103465] 2022/10/18 15:03:06.072013 [ERR] 168.63.70.231:47496 - rid:108 - Error flushing: writev tcp 168.63.70.235:4248->168.63.70.231:47496: writev: connection reset by peer
[103465] 2022/10/18 15:03:06.091311 [INF] 168.63.70.231:47496 - rid:108 - Router connection closed: Write Error
[103465] 2022/10/18 15:03:06.635893 [INF] 168.63.70.231:46632 - rid:16794 - Route connection created
[103465] 2022/10/18 15:03:07.455360 [DBG] 168.63.70.231:46632 - rid:16794 - Registering remote route "NDYJNRGBH5WOIPOO3DYACVPPJCARLSREVZCR5AUUMJK5KBXDQZ7Q44CI"
[103465] 2022/10/18 15:03:07.464842 [DBG] Not attempting reconnect for solicited route, already connected to "NDYJNRGBH5WOIPOO3DYACVPPJCARLSREVZCR5AUUMJK5KBXDQZ7Q44CI"
[103465] 2022/10/18 15:03:31.552958 [DBG] 168.63.70.231:46632 - rid:16794 - Sent local subscriptions to route
[103465] 2022/10/18 15:03:31.553606 [WRN] 168.63.70.231:46632 - rid:16794 - Readloop processing time: 24.916582459s
[103465] 2022/10/18 15:03:31.577474 [DBG] 168.63.70.231:46632 - rid:16794 - Router Ping Timer

[103465] 2022/10/18 15:03:31.605826 [ERR] 168.63.70.231:46632 - rid:16794 - Error flushing: writev tcp 168.63.70.235:4248->168.63.70.231:46632: writev: connection reset by peer
[103465] 2022/10/18 15:03:31.605852 [INF] 168.63.70.231:46632 - rid:16794 - Router connection closed: Write Error
[103465] 2022/10/18 15:03:31.608639 [DBG] 168.63.70.231:46632 - rid:16794 - Removing 3 subscriptions for account "$G"
[103465] 2022/10/18 15:04:10.135900 [DBG] 168.63.70.231:47496 - rid:108 - Removing 467969 subscriptions for account "$G"
```
we count found the first route closed is 15:03:05.855381,but soon go on the reconnect;  
and the second route closed is 15:03:22.174885,but never reconnect; 
we want to know why?

**Comment by domdom82** [comment]
Hi @wubumihuo this is indeed strange behavior. I've noticed you are using a pretty outdated NATS version 2.1.7
The latest [nats-release](https://github.com/cloudfoundry/nats-release/releases/tag/v56.12.0) for Cloudfoundry is v2.10.4. So the first step I would take it to bump the release to the latest version.

We've encounted issues with reconnects with the introduction of a connection pool in v2.10 and have [patched it out](https://github.com/cloudfoundry/nats-release/pull/67) for the latest release. This shouldn't affect you, however as your version is even older.

What I saw in your config is that the nodes don't share the same sets of routes in the cluster. This is odd, they should all have the same configuration there. I don't know if that is the root cause but it looks suspicious.
